src/plots/smith/smith.js

```diff
@@ -0,0 +1,1115 @@
+                r0 = rr1;
+                r1 = rr0;
+            }
+            return true;
+        } else {
+            r0 = null;
+            r1 = null;
+            return false;
+        }
+    }
+
+    function applyZoomMove(path1, cpath) {
+        path1 = path1 || path0;
+        cpath = cpath || 'M0,0Z';
+
+        zb.attr('d', path1);
+        corners.attr('d', cpath);
+        dragBox.transitionZoombox(zb, corners, dimmed, lum);
+        dimmed = true;
+
+        var updateObj = {};
+        computeZoomUpdates(updateObj);
+        gd.emit('plotly_relayouting', updateObj);
+    }
+
+    function zoomMove(dx, dy) {
+        dx = dx * scaleX;
+        dy = dy * scaleY;
+
+        var x1 = x0 + dx;
+        var y1 = y0 + dy;
+
+        var rr0 = xy2r(x0, y0);
+        var rr1 = Math.min(xy2r(x1, y1), radius);
+        var a0 = xy2a(x0, y0);
+        var path1;
+        var cpath;
+
+        if(clampAndSetR0R1(rr0, rr1)) {
+            path1 = path0 + _this.pathSector(r1);
+            if(r0) path1 += _this.pathSector(r0);
+            // keep 'starting' angle
+            cpath = pathCorner(r0, a0) + pathCorner(r1, a0);
+        }
+        applyZoomMove(path1, cpath);
+    }
+
+    function findPolygonRadius(x, y, va0, va1) {
+        var xy = helpers.findIntersectionXY(va0, va1, va0, [x - cxx, cyy - y]);
+        return norm(xy[0], xy[1]);
+    }
+
+    function zoomMoveForPolygons(dx, dy) {
+        var x1 = x0 + dx;
+        var y1 = y0 + dy;
+        var a0 = xy2a(x0, y0);
+        var a1 = xy2a(x1, y1);
+        var vangles0 = findEnclosingVertexAngles(a0, vangles);
+        var vangles1 = findEnclosingVertexAngles(a1, vangles);
+        var rr0 = findPolygonRadius(x0, y0, vangles0[0], vangles0[1]);
+        var rr1 = Math.min(findPolygonRadius(x1, y1, vangles1[0], vangles1[1]), radius);
+        var path1;
+        var cpath;
+
+        if(clampAndSetR0R1(rr0, rr1)) {
+            path1 = path0 + _this.pathSector(r1);
+            if(r0) path1 += _this.pathSector(r0);
+            // keep 'starting' angle here too
+            cpath = [
+                pathCornerForPolygons(r0, vangles0[0], vangles0[1]),
+                pathCornerForPolygons(r1, vangles0[0], vangles0[1])
+            ].join(' ');
+        }
+        applyZoomMove(path1, cpath);
+    }
+
+    function zoomDone() {
+        dragBox.removeZoombox(gd);
+
+        if(r0 === null || r1 === null) return;
+        var updateObj = {};
+        computeZoomUpdates(updateObj);
+
+        dragBox.showDoubleClickNotifier(gd);
+
+        Registry.call('_guiRelayout', gd, updateObj);
+    }
+
+    function computeZoomUpdates(update) {
+        var rl = radialAxis._rl;
+        var m = (rl[1] - rl[0]) / (1 - innerRadius / radius) / radius;
+        var newRng = [
+            rl[0] + (r0 - innerRadius) * m,
+            rl[0] + (r1 - innerRadius) * m
+        ];
+        update[_this.id + '.realaxis.range'] = newRng;
+    }
+
+    function zoomClick(numClicks, evt) {
```

Comment:
Is this function used?

Reply:
8351bfa163c02d14ca5a1c6e8c5ae0d1520a9673